tuf/repository_lib.py

```diff
@@ -1535,9 +1555,235 @@ def _get_hashes_and_length_if_needed(use_length, use_hashes, full_file_path,
 
 
 
+class Node(object):
+  """
+  Merkle tree node that keeps track of the node hash and the parent node.
+  """
+  _parent = None
+  _hash = None
+
+  def __init__(self):
+    return
+
+  def parent(self):
+    return self._parent
+
+  def set_parent(self, parent):
+    self._parent = parent
+
+  def hash(self):
+    return self._hash
+
+
+
+class InternalNode(Node):
+  """
+  An internal Merkle tree node that keeps track of a left and a right
+  child. Upon creation, this node takes in a left and right Node
+  and computes the hash of (left + right). In addition, the constructor
+  sets the parent node of left and right to this node to allow for
+  traversal of the tree.
+  """
+  _left = None
+  _right = None
+
+  def __init__(self, left, right):
+    super(InternalNode, self).__init__()
+    self._left = left
+    self._right = right
+
+    left.set_parent(self)
+    right.set_parent(self)
+    digest_object = securesystemslib.hash.digest(algorithm=HASH_FUNCTION)
+
+    digest_object.update((left.hash() + right.hash()).encode('utf-8'))
+
+    self._hash = digest_object.hexdigest()
+
+  def left(self):
+    return self._left
+
+  def right(self):
+    return self._right
+
+  def isLeaf(self):
+    return False
+
+
+
+class Leaf(Node):
+  """
+  This Merkle tree leaf node keeps track of the node contents and name.
+  The name should correspond with a metadata file and the contents should
+  contain the snapshot information for that metadata file.
+
+  The constructor takes in a name and contents and computes the hash
+  of the contents. The hash may be provided to save computation time
+  if it has already been computed.
+  """
+  # Merkle Tree leaf
+  _contents = None
+  _name = None
+
+  def __init__(self, name, contents, digest = None):
```

Comment:
```suggestion
  def __init__(self, name, contents, hash=None):
```
No need for spaces around the assignment operator for default parameters. I think it's slightly clearer if we call this optional parameter hash, as that's the internal variable name and the term used in the docstring.